src/internal/components/InternalEditor.tsx

```diff
@@ -97,41 +96,62 @@ export const InternalEditor = ({
   };
 
   const handleSave = async (data: Data) => {
+    if (themeModeActive) {
+      devLogger.logFunc("saveStyles");
+      // save styles
+      return;
+    }
     devLogger.logFunc("saveVisualConfigData");
     saveVisualConfigData({
       payload: { visualConfigurationData: JSON.stringify(data) },
     });
   };
 
-  const change = async () => {
-    if (isLoading) {
-      return;
-    }
-    if (!canEdit) {
-      setCanEdit(true);
-      return;
-    }
+  const toggleThemeModeActive = () => {
+    setThemeModeActive((prev) => !prev);
   };
 
+  // set permissions on the component level to allow for dynamic updating
+  Object.values(puckConfig.components).forEach((component) => {
```

Comment:
Does this need to be in a useEffect? If themeModeActive changes does this update properly? Or does refreshPermissions somehow just rerender this component?

Reply:
This works because the themeModeActive state is defined in this component, so it rerenders when that updates.

If I put it in a useEffect with [themeModeActive], it unfortunately doesn't work because the header useEffect that calls refreshPermissions fires first.

But I'm not sure -- is it better to have this in a useEffect and have it set another state variable when it completes, which then triggers the header useEffect?

Reply:
If it's working and not causing a bunch of rerenders that fire other things off needlessly then it's fine. You can use the xYextDebug=true param to see if we're now making more calls to things before/after.

Reply:
It didn't seem to be making more calls, but I realized it works if I moved it to the header useEffect so that seems better

Reply:
nice